web/docs/project/env-vars.md

```diff
@@ -240,45 +376,101 @@ console.log(env.REACT_APP_SOME_VAR_NAME)
 ```ts title="src/App.ts"
 import { env } from 'wasp/client'
 
-console.log(env.REACT_APP_SOME_VAR_NAME)
+// Wasp-defined
+const apiUrl: string = env.REACT_APP_API_URL
+
+// User-defined
+const analyticsId: string = env.REACT_APP_ANALYTICS_ID
 ```
 
 </TabItem>
 </Tabs>
 
-The `env` object is a validated object that Wasp provides to access client env vars.
-
 You can use `import.meta.env.REACT_APP_SOME_VAR_NAME` directly in your code, but it's not recommended because it's not validated and can lead to runtime errors if the env var is not defined.
 
-<!-- TODO: when the Zod validation PRs are merged, describe how users can define their own validations -->
-
 ### Server Env Vars
 
-Access server env vars in your server code using the `env` object like this:
+#### User-defined env vars validation
+
+You can define your env vars validation like this:
 
 <Tabs groupId="js-ts">
 <TabItem value="js" label="JavaScript">
 
-```js title="src/App.js"
+```js title="src/env.js"
+import * as z from 'zod'
+
+export const envValidationSchema = z.object({
+  STRIPE_API_KEY: z.string({
+    required_error: 'STRIPE_API_KEY is required.',
+  }),
+})
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```ts title="src/env.ts"
+import * as z from 'zod'
+
+import { defineEnvValidationSchema } from 'wasp/env'
+
+export const envValidationSchema = defineEnvValidationSchema(
+  z.object({
+    STRIPE_API_KEY: z.string({
+      required_error: 'STRIPE_API_KEY is required.',
+    }),
+  })
+)
+```
+
+We are using the `defineEnvValidationSchema` helper to get type-checking for the Zod schema.
+
+</TabItem>
+</Tabs>
+
+```wasp title="main.wasp"
+app myApp {
+  ...
+  server: {
+    envValidationSchema: import { envValidationSchema } from "@src/env",
+  },
+}
+```
+
+Wasp merges your env validation schemas with the built-in env vars validation schemas when it validates the `process.env` object.
+
+#### Accessing env vars in server code
+
+You can access both **Wasp-defined** and **user-defined** client env vars in your client code using the `env` object:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```js title="src/stripe.js"
 import { env } from 'wasp/server'
 
-console.log(env.SOME_SECRET)
+// Wasp-defined
+const serverUrl = env.WASP_SERVER_URL
+
+// User-defined
+const stripeApiKey = env.STRIPE_API_KEY
 ```
 
 </TabItem>
 <TabItem value="ts" label="TypeScript">
 
-```ts title="src/App.ts"
+```ts title="src/stripe.ts"
 import { env } from 'wasp/server'
 
-console.log(env.SOME_SECRET)
+// Wasp-defined
+const serverUrl: string = env.WASP_SERVER_URL
+
+// User-defined
+const stripeApiKey: string = env.STRIPE_API_KEY
 ```
 
 </TabItem>
 </Tabs>
 
-The `env` object is a validated object that Wasp provides to access server env vars.
-
 You can use `process.env.SOME_SECRET` directly in your code, but it's not recommended because it's not validated and can lead to runtime errors if the env var is not defined.
```

Comment:
Same suggestion as above. This sentence is kind of a mouthful (that's what she said).